assets/sequences/sfxbanks/voice.seq.inc

```diff
@@ -0,0 +1,1083 @@
+ldptr addr_6499
+jump play_single_voice
+
+.array addr_6499
+byte FONT00_EFFECT_LINK_CHILD_ATTACK_0
+byte 100
+
+.channel voice_link_child_shiver
+ldlayer 0, layer_649F
+end
+
+.layer layer_649F
+transpose 1
+notedv PITCH_C1, 0x0, 50
+end
+
+.channel voice_link_child_jump
+ldio 6
+sub 255
+stio 6
+and 1
+ldseq addr_64C5
+stseq 64, layer_64BF
+rand 2
+ldseq addr_64C7
+stseq 0, layer_64BF+2
+ldlayer 0, layer_64BF
+end
+
+.layer layer_64BF
+notedv PITCH_A0, 0x0, 100
+ldelay 0x30
+end
+
+.array addr_64C5
+byte FONT00_EFFECT_LINK_CHILD_JUMP_0
+byte FONT00_EFFECT_LINK_ADULT_JUMP_1
+
+.array addr_64C7
+byte 80
+byte 80
+
+.channel voice_link_child_cast_spell1
+ldptr addr_64CF
+jump play_single_voice
+
+.array addr_64CF
+byte FONT00_EFFECT_LINK_CHILD_STRONG_1
+byte 110
+
+.channel voice_link_child_shock
+ldlayer 0, layer_64D5
+end
+
+.layer layer_64D5
+transpose 2
+notedv PITCH_E1, 0x0, 85
+end
+
+.channel voice_link_child_cast_spell2
+ldptr addr_64E1
+jump play_single_voice
+
+.array addr_64E1
+byte FONT00_EFFECT_LINK_CHILD_STRONG_0
+byte 95
+
+.channel voice_link_child_push_block
+ldptr addr_64E9
+jump play_single_voice
+
+.array addr_64E9
+byte FONT00_EFFECT_LINK_CHILD_SIGH_0
+byte 90
+
+.channel voice_link_child_hookshot_hang
+ldptr addr_64F1
+jump play_single_voice
+
+.array addr_64F1
+byte FONT00_EFFECT_LINK_CHILD_LEDGE_0
+byte 100
+
+.channel voice_link_child_knockback
+ldptr addr_64F9
+jump play_single_voice
+
+.array addr_64F9
+byte FONT00_EFFECT_LINK_CHILD_FALL_DMG
+byte 115
+
+.channel voice_link_child_unused
+ldptr addr_6501
+jump play_single_voice
+
+.array addr_6501
+byte FONT00_EFFECT_LINK_CHILD_CHARGE
+byte 100
+
+.channel voice_link_child_cast_spell3
+ldptr addr_6509
+jump play_single_voice
+
+.array addr_6509
+byte FONT00_EFFECT_LINK_CHILD_SPELL
+byte 110
+
+.channel voice_link_child_cutscene_attacked
+ldptr addr_6511
+jump play_single_voice
+
+.array addr_6511
+byte FONT00_EFFECT_LINK_CHILD_FALL_1
+byte 100
+
+.channel voice_navi_greeting
+ldi 32
+stio 1
+ldlayer 0, layer_651A
+end
+
+.layer layer_651A
+transpose 2
+notedv PITCH_DF1, 0x0, 100
+end
+
+.channel voice_npcs
+ldio 4
+sub 80
+stseq 25, layer_6536  # set semitone = sound id - 80 + 25
+ldseq addr_653B
+stseq 0, layer_6536+2
+ldlayer 0, layer_6534
+panweight 96
+end
+
+.layer layer_6534
+transpose 1
+.layer layer_6536
+notedvg 33, 0x0, 100, 0
+end
+
+.array addr_653B
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 110
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 100
+byte 110
+
+.channel voice_other_npcs
+ldio 4
+sub 119
+stseq 0, layer_6576  # set semitone = sound id - 119
+ldseq addr_657B
+stseq 0, layer_6576+2
+ldlayer 0, layer_6574
+end
+
+.layer layer_6574
+transpose 2
```

Comment:
In the "fonts", there are three distinct types of sound: "Instruments", "drums", "sound effects".

Currently, `semitone` and `transpose` are named after how they are used for "Instruments", and they are decent name.

However, these two variables have completely different meaning when it comes to sound effects (this particular code is part of the infamous "Mweep" sfx").

"semitone" for sound effects actually stores the lower bit of the sfxId (the index to the list of soundeffects in the font, and NOT the id's for sfx that are exposed externally to audio)

"transpose" for sound effects stores the upper bit of the sfxId.

You can see the code for this here (which needs updated docs, probably unions to represent this different functionality):
https://github.com/zeldaret/oot/blob/9f09505d34619883748a7dab05071883281c14fd/src/code/audio_seqplayer.c#L833

Now regarding the instructions here, would it make sense to have different named commands representing the same OP instruction, since it has different meaning depending on if it's processing an instrument/drum/sound effect?

Reply:
Particularly, `semitone` and `transpose` are set so that it constructs this enum value:
https://github.com/MNGoldenEagle/oot/blob/fb5bcb9e5e9bfa5e9735c9a3015aa32d896a1767/assets/xml/soundfonts/Sound%20Effects%201.xml#L235

Here's the steps below on how it achieves that:

There's the "external" mweep sfxId:
https://github.com/zeldaret/oot/blob/9f09505d34619883748a7dab05071883281c14fd/include/tables/sfx/voicebank_table.h#L135
Which the table entry ("sound id as you have in the comment above") is extracted from that via the mask `& 0x1FF`, i.e. `0x7A` or 122.

Then, `stseq 0, layer_6576  # set semitone = sound id - 119` is evaluated to 3

So the `semitone = 3` and `transposition = 2`

Those are the 2 values needed to construct the internal sfx:
`sfxId = (layer->transposition << 6) + semitone;`
`sfxId = (2 << 6) + 3;`
`sfxId = 131`

Looking back at:
https://github.com/MNGoldenEagle/oot/blob/fb5bcb9e5e9bfa5e9735c9a3015aa32d896a1767/assets/xml/soundfonts/Sound%20Effects%201.xml#L235
The `<Effect Name="King Zora Scoots" Enum="MWINK"/>` is the 131'th sound effect entry in that font. In fact, the enum `MWINK` is valued at 131. So the `sfxId = 131` or `sfxId = MWINK` is set to get that entry.

So is there any way to have `transposition` and `semitone` set so that it uses the enum `MWINK` and not have it hardcoded? I don't see any easy solution but just wanted to walk through things

Reply:
TBD. Interesting stuff though.